### ISSUE TITLE
Foreign key constraints are not added to the tables

### ISSUE THREAD

**Issue by vinkaga** [balderdashy/sails-mysql#172]
Using an example from waterline (express-example.js), which contains the following model
```JavaScript
var Pet = Waterline.Collection.extend({

    identity: 'pet',
    connection: 'db',

    attributes: {
        name: 'string',
        breed: 'string',
        // Associations
        userId: {
            model: 'user'
        }
    }
});
```
The query generated for this model is
```SQL
CREATE TABLE `pet` (`name` VARCHAR(255) , `breed` VARCHAR(255) , `userId` INT , `id` INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY, `createdAt` DATETIME , `updatedAt` DATETIME )
```
The query should contain foreign key constraint for userId but doesn't.

**Comment by pballester** [comment]
Can anyone help? I have the same issue.

Thanks.

**Comment by Zephyrante** [comment]
Same issue here, any hints?

Thx

**Comment by ti0ma** [comment]
I have the same issue, any patch?

Cheers!

**Comment by fonseca-ro** [comment]
waterline doesn´t create any constraint foreign for you, if you want it, you´ve to create in your database.

**Comment by fonseca-ro** [comment]
what waterline does is just "join" your tables.

**Comment by hardchor** [comment]
Please correct me if I'm wrong, but that's a bit of a shortcoming. How would you manage associations, cascading deletes etc. without foreign keys?

**Comment by randallmeeker** [comment]
https://github.com/balderdashy/waterline/issues/794

**Comment by tejeshm** [comment]
Just tried waterline and realized the issue. This could be because it handle all the databases sql or nosql equally, but it would be great if it can determine by the connection type and create the foreign key for relational databases, or at least give an option to explicitly mention it.  Because manually creating only the foreign key is not such a good idea.

**Comment by nathanmarks** [comment]
+1 on this.

It would be great to be able to setup proper foreign keys, cascading deletes, etc etc... It's the most annoying part of using waterline+mysql so far.

**Comment by gunish** [comment]
My understanding on why this is not done, because waterline allows us to have separate adapters for individual models, and it would be impossible to build cross database relationships, so I think the creators completely left it out. 

It would have been cool, if waterline to detect that everything was going into the same model and implement the schema constraints, if asked to do via a adapter setting ? 

This might need some major rework though, so for now, we let waterline create our tables and then we implement the join and constraints manually, (by a script) and we then leave ```migrate : safe``` in production.

**Comment by argordmel** [comment]
I have the same issue.

Thanks.

**Comment by cesardeazevedo** [comment]
+1, its would be awesome,

i've moved forward with sequelize instead, but since the sails blueprints have saved me a LOT of time, i've did a blueprints hook to work with sequelize.

[https://github.com/cesardeazevedo/sails-hook-sequelize-blueprints](https://github.com/cesardeazevedo/sails-hook-sequelize-blueprints)

**Comment by sailsbot** [comment]
Thanks for posting, @vinkaga.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by apinnecke** [comment]
Having foreign keys would be really nice, though. Even if it's not able to support them on all Databases, this is a specialized repo just for mysql, so I don't understand the problem adding foreign keys?

+1 to the post of @gunish. @vinkaga would you please reopen this issue?
